pkg/util/net.go

```diff
@@ -97,10 +97,9 @@ func GetFirst6(elems []net.IP) (net.IP, error) {
 // If no IPv6 addresses are found, an error is raised.
 func GetFirst6Net(elems []*net.IPNet) (*net.IPNet, error) {
 	for _, elem := range elems {
-		if elem == nil || elem.IP.To16() == nil {
-			continue
+		if elem != nil || netutils.IsIPv6(elem.IP) {
```

Comment:
This is wrong, you'll now return any non-nil element!

```suggestion
		if elem != nil && netutils.IsIPv6(elem.IP) {
```

Also, note that the code you're calling effectively does the same thing - checks for non-nil `.To16()`:
https://github.com/kubernetes/utils/blob/9f6742963106fb1fadd7ec2493d6d6cd689e5318/net/ipfamily.go#L106

Reply:
good catch! I missed that :facepalm: 

It is slightly different in the ipfamily.go code because by the time to code reaches `case ip.To16() != nil:`, ipv4  ips were already returned in the previous line, i.e. only ipv6 addresses can reach `case ip.To16() != nil:`

Reply:
Run this gist and you'll see the difference: https://gist.github.com/manuelbuil/73d7e0ff32a69eb045602aeee45896a6